File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.cpp
class Solution {
public:
    vector<char> vowels = { 'a', 'e', 'i', 'o', 'u'};
    bool isVowelString(string s){
        int c=0;
        for(auto v:vowels){
            if(v==s.front()) c++;
            if(v==s.back()) c++;
        }
        return c==2;
    }
    
    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& queries) {
        vector<int> v = {0};
        for(auto w:words)
            v.push_back(isVowelString(w)?1:0);
        for(int i=1; i<v.size(); i++)
            v[i] += v[i-1];
        v.push_back(v.back());
        vector<int> ans;
        for(auto q: queries)
            ans.push_back(v[q[1]+1]-v[q[0]]);
        return ans;
    }
};

File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.java
class Solution {
    private static Set<Character> vowels = new HashSet<>(Arrays.asList('a', 'e', 'i', 'o', 'u'));

    public int[] vowelStrings(String[] words, int[][] queries) {
        int[] vowelFlankedWords = new int[words.length], ans = new int[queries.length];
        int prefixSum = 0;
        for (int i = 0; i < words.length; i++) {
            String s = words[i];
            vowelFlankedWords[i] = prefixSum;
            if (vowels.contains(s.charAt(0))
                    && vowels.contains(s.charAt(s.length() - 1)))
                prefixSum = vowelFlankedWords[i] += 1;
        }
        for (int i = 0; i < queries.length; i++) {
            int[] q = queries[i];
            ans[i] = q[0] == 0 ? vowelFlankedWords[q[1]] : vowelFlankedWords[q[1]] - vowelFlankedWords[q[0] - 1];
        }
        return ans;
    }
}